src/Mealz/AccountingBundle/Tests/Controller/CostSheetControllerTest.php

```diff
@@ -4,56 +4,69 @@
 use App\Mealz\AccountingBundle\Entity\Transaction;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadCategories;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadDays;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadDishes;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadDishVariations;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadMeals;
-use App\Mealz\MealBundle\DataFixtures\ORM\LoadWeeks;
+use App\Mealz\AccountingBundle\Service\Wallet;
+use App\Mealz\MealBundle\Entity\Participant;
 use App\Mealz\MealBundle\Tests\Controller\AbstractControllerTestCase;
 use App\Mealz\UserBundle\DataFixtures\ORM\LoadRoles;
 use App\Mealz\UserBundle\DataFixtures\ORM\LoadUsers;
+use DateTime;
 
 /**
  * Class CostSheetControllerTest.
  */
 class CostSheetControllerTest extends AbstractControllerTestCase
 {
+    private Wallet $wallet;
+
     protected function setUp(): void
     {
         parent::setUp();
 
         $this->clearAllTables();
         $this->loadFixtures([
-            new LoadWeeks(),
-            new LoadDays(),
-            new LoadCategories(),
-            new LoadDishes(),
-            new LoadDishVariations(),
-            new LoadMeals(),
             new LoadRoles(),
             new LoadUsers(self::$container->get('security.user_password_encoder.generic')),
             new LoadTransactions(),
         ]);
 
+        $participantRepo = $this->getDoctrine()->getRepository(Participant::class);
+        $transactionRepo = $this->getDoctrine()->getRepository(Transaction::class);
+        $this->wallet = new Wallet($participantRepo, $transactionRepo);
+
         $this->loginAs(self::USER_KITCHEN_STAFF);
     }
 
     /**
      * Check if hashCode was written in database.
      */
-    public function testHashWrittenInDatabase(): void
+    public function testHashWrittenInDatabaseWithBalance(): void
     {
         $profile = $this->getUserProfile(self::USER_STANDARD);
-
         $this->assertNull($profile->getSettlementHash(), 'SettlementHash was set already');
 
-        $this->client->request('GET', '/print/costsheet/settlement/request/alice.meals');
+        $balance = $this->wallet->getBalance($profile);
+        $this->assertGreaterThan(0.00, $balance);
+
+        $this->client->request('GET', '/print/costsheet/settlement/request/' . $profile->getUsername());
         $this->assertTrue($this->client->getResponse()->isSuccessful());
 
         $this->assertNotNull($profile->getSettlementHash(), 'SettlementHash was not set');
     }
 
+    public function testHashWrittenInDatabaseWithoutBalance(): void
+    {
+        $profile = $this->getUserProfile('john.meals');
+        $this->assertNull($profile->getSettlementHash(), 'SettlementHash was set already');
+
+        $balance = $this->wallet->getBalance($profile);
+        $this->assertEquals(0.00, $balance);
+
+        $this->client->request('GET', '/print/costsheet/settlement/request/' . $profile->getUsername());
+        $this->assertTrue($this->client->getResponse()->isSuccessful());
```

Comment:
Can be simplified to "self::assertResponseIsSuccessful();".

Reply:
Done.